BWEnv/src/utils.cc

```diff
@@ -245,8 +245,11 @@ std::string readIni(const std::string& filename, const std::string& section, con
   long filesize = ftell(f);
   data.resize(filesize);
   fseek(f, 0, SEEK_SET);
-  fread(data.data(), filesize, 1, f);
+  auto elementsRead = fread(data.data(), filesize, 1, f);
   fclose(f);
+  if (!elementsRead) {
+    return {};
+  }
```

Comment:
Unrelated to serialization, but addresses GCC warning for disposed return value.

Reply:
https://stackoverflow.com/questions/689677/why-cast-unused-return-values-to-void

It looks like the idiomatic way to do this is `(void) fread`?

Reply:
That silences the warning, but doesn't address the reason why the warning exists in the first place: Reading a file can fail, and it's a case you probably want to handle. In this case, there's a perfectly good default, though I should probably add a log warning.

Reply:
Yep, best practice is to check for errors here; however, this should be done by `if (elementsRead != filesize)`.